### PR TITLE
Enable MQTT over websockets support

### LINKED ISSUE #41: Allow mqtt over ws
I guess we can add a 'protocol' field in the mqtt config, and sdk could use that protocol instead of hardcoded 'mqtt' or 'mqtts'.

### PR DESCRIPTION
**Pull request by oguzcankirmemis**
- MQTT config now accepts an additional "protocol" property to specify
  which protocol should be used to connect.
- If "protocol" is not specified, it fallbacks to usual protocol "mqtt"
  or "mqtts", depending on whether the "secure" property is set.

resolves #41

Signed-off-by: Oguzcan Kirmemis <oguzcan.kirmemis@gmail.com>

### CODE REVIEW

File: api/mqtt/connector.js [hunk 1]
```diff
@@ -54,18 +55,18 @@ function Broker(conf) {
...
-                        password: me.deviceInfo.device_token,
-                        rejectUnauthorized: false
-                    });
+                var protocol = 'mqtt://';
+                if (me.secure) {
+                    protocol = 'mqtts://'
+                }
+                if (me.protocol) {
+                    protocol = me.protocol + '://';
                 }
```
Comment (srware) [inline api/mqtt/connector.js:64]: I think we should do some sanity checking here to make sure protocol isn't something stupid. I assume there should be 3 valid options, mqtt, mqtts and ws?

**Review by srware — CHANGES_REQUESTED** [review]
I think I would prefer to add a config option called 'websockets' to make this more robust. If 'websockets' is false or doesn't exist just use mqtt or mqtts as before. The protocol config property opens up a lot of potential for failure or worse.

File: api/mqtt/connector.js (hunk 1 above)
Comment (srware) [reply api/mqtt/connector.js:64]: You can ignore this if you agree with my comments above :)

**Comment by oguzcankirmemis** [comment]
Technically speaking, mqtt allows other protocols than websockets, and that was on my mind. Nevertheless, emqx does not support those and we also do not have plans to support them. So, I agree that it makes things unnecessarily complex and have changed it to use "websockets".

**Review by srware — APPROVED** [review]

**Pull request merged**